mypyc/test-data/irbuild-basic.test

```diff
@@ -3546,3 +3546,420 @@ L0:
     r2 = PyObject_Vectorcall(r1, 0, 0, 0)
     r3 = box(None, 1)
     return r3
+
+[case testForMapBasic]
+def f(x: int) -> int:
+    return x * 2
+def g(a: list[int]) -> int:
+    s = 0
+    for x in map(f, a):
+        s += x
+    return s
+[out]
+def f(x):
+    x, r0 :: int
+L0:
+    r0 = CPyTagged_Multiply(x, 4)
+    return r0
+def g(a):
+    a :: list
+    s :: int
+    r0 :: dict
+    r1 :: str
+    r2 :: object
+    r3, r4 :: native_int
+    r5 :: bit
+    r6 :: object
+    r7, _mypyc_map_arg_0 :: int
+    r8 :: object
+    r9, r10, x, r11 :: int
+    r12 :: native_int
+L0:
+    s = 0
+    r0 = __main__.globals :: static
+    r1 = 'f'
+    r2 = CPyDict_GetItem(r0, r1)
+    r3 = 0
+L1:
+    r4 = var_object_size a
+    r5 = r3 < r4 :: signed
+    if r5 goto L2 else goto L4 :: bool
+L2:
+    r6 = list_get_item_unsafe a, r3
+    r7 = unbox(int, r6)
+    _mypyc_map_arg_0 = r7
+    r8 = box(int, _mypyc_map_arg_0)
+    r9 = unbox(int, r8)
+    r10 = f(r9)
+    x = r10
+    r11 = CPyTagged_Add(s, x)
+    s = r11
+L3:
+    r12 = r3 + 1
+    r3 = r12
+    goto L1
+L4:
+L5:
+    return s
+
+[case testForMapMultiple]
+from typing import Any
+def f(x: int, y: int) -> int:
+    return x * y
+def g(a: list[int], b: list[int]) -> int:
+    s = 0
+    for x in map(f, a, b):
+        s += x
+    return s
+[out]
+def f(x, y):
+    x, y, r0 :: int
+L0:
+    r0 = CPyTagged_Multiply(x, y)
+    return r0
+def g(a, b):
+    a, b :: list
+    s :: int
+    r0 :: dict
+    r1 :: str
+    r2 :: object
+    r3, r4, r5 :: native_int
+    r6 :: bit
+    r7 :: native_int
+    r8 :: bit
+    r9 :: object
+    r10, _mypyc_map_arg_0 :: int
+    r11 :: object
+    r12, _mypyc_map_arg_1 :: int
+    r13, r14 :: object
+    r15, r16, r17, x, r18 :: int
+    r19, r20 :: native_int
+L0:
+    s = 0
+    r0 = __main__.globals :: static
+    r1 = 'f'
+    r2 = CPyDict_GetItem(r0, r1)
+    r3 = 0
+    r4 = 0
+L1:
+    r5 = var_object_size a
+    r6 = r3 < r5 :: signed
+    if r6 goto L2 else goto L5 :: bool
+L2:
+    r7 = var_object_size b
+    r8 = r4 < r7 :: signed
+    if r8 goto L3 else goto L5 :: bool
+L3:
+    r9 = list_get_item_unsafe a, r3
+    r10 = unbox(int, r9)
+    _mypyc_map_arg_0 = r10
+    r11 = list_get_item_unsafe b, r4
+    r12 = unbox(int, r11)
+    _mypyc_map_arg_1 = r12
+    r13 = box(int, _mypyc_map_arg_0)
+    r14 = box(int, _mypyc_map_arg_1)
+    r15 = unbox(int, r13)
+    r16 = unbox(int, r14)
+    r17 = f(r15, r16)
+    x = r17
+    r18 = CPyTagged_Add(s, x)
+    s = r18
+L4:
+    r19 = r3 + 1
+    r3 = r19
+    r20 = r4 + 1
+    r4 = r20
+    goto L1
+L5:
+L6:
+    return s
+
+[case testForMapThreeArgs]
```

Comment:
The number of irbuild tests is perhaps a bit much? I wonder if 1-2 irbuild test cases would be sufficient, with the remaining cases only covered by run tests.

Reply:
I picked 3 to keep, one-arg, 3-arg, and comprehension